src/fidesops/service/connectors/sql_connector.py

```diff
@@ -357,6 +359,32 @@ def create_client(self) -> Engine:
             echo=not self.hide_parameters,
         )
 
+    # Overrides SQLConnector.query_config
+    def query_config(self, node: TraversalNode) -> BigQueryQueryConfig:
+        """Query wrapper corresponding to the input traversal_node."""
+        return BigQueryQueryConfig(node)
+
+    def mask_data(
+        self,
+        node: TraversalNode,
+        policy: Policy,
+        request: PrivacyRequest,
+        rows: List[Row],
+    ) -> int:
+        """Execute a masking request. Returns the number of records masked"""
+        query_config = self.query_config(node)
+        update_ct = 0
+        client = self.client()
+        for row in rows:
+            update_stmt: Optional[Executable] = query_config.generate_update(
+                row, policy, request, client
+            )
```

Comment:
Hmm, I've gone back and forth on this, whether a new "generate_update" method is more clear, as you've demonstrated, or alternatively, we override `BigQueryQueryConfig.generate_update_stmt` to return a Sqlalchemy Update object, which can be executed just like a TextClause, so we don't have to override `mask_data`.   And we have `generate_update_stmt` types to expand to be any type of Executable. What are your thoughts?

Reply:
My thinking is `generate_update_stmt` was misleading naming in the first place, since it returned a `Optional[TextClause]` instead of using the `update` method of sqlalchemy. So here I'd like to delineate more obviously from that TextClause than an override provides. 

The main thing, though, is we need an additional param, `client`, passed in to `generate_update` to generate the Update object for BigQuery (as opposed to the base `generate_update_stmt` which does not require `client`).

Reply:
hm, yeah good point